habitat/hooks/run

```diff
@@ -2,59 +2,55 @@
 exec 2>&1
 
 # Set credentials
-TPP_URL="{{ cfg.tpp.url }}"
-TPP_USER="{{ cfg.tpp.user }}"
-TPP_PASSWORD="{{ cfg.tpp.password }}"
-COMMON_NAME="{{ cfg.tpp.cn }}"
-CERTIFICATE_ZONE="{{ cfg.tpp.zone }}"
-
-# Set cert file
+{{#unless cfg.tpp.auth.url ~}}
+TPP_URL="{{ cfg.tpp.auth.url }}"
+TPP_USER="{{ cfg.tpp.auth.user }}"
+TPP_PASSWORD="{{ cfg.tpp.auth.password }}"
+{{/unless}}
+COMMON_NAME="{{ cfg.cn }}"
+{{#unless cfg.cloud.auth.apikey ~}}
+CLOUD_APIKEY="{{ cfg.cloud.auth.apikey }}"
+{{/unless}}
+
+# Set device information
+APP_INFO="{{ cfg.tpp.device.app-info }}"
+INSTANCE="{{ sys.hostname }}{{#if cfg.tpp.device.app-name ~}}:{{ cfg.tpp.device.app-name }}{{/if ~}}"
+{{#unless cfg.tpp.device.tls_address ~}}
+TLS_ADDRESS="{{ sys.ip }}:{{ cfg.tpp.device.tls_port }}"
+{{/unless}}
+{{#if cfg.tpp.device.tls_address ~}}
+TLS_ADDRESS="{{ cfg.tpp.device.tls_address }}:{{ cfg.tpp.device.tls_port }}"
+{{/if ~}}
+
+# Set certificate file names
 CERT_FILE="${COMMON_NAME}.cert"
 KEY_FILE="${COMMON_NAME}.key"
 CHAIN_FILE="${COMMON_NAME}.chain"
+ID_FILE="${COMMON_NAME}.id"
 
 # Set cert path
 CERT_PATH="{{pkg.svc_data_path}}/${CERT_FILE}"
 KEY_PATH="{{pkg.svc_data_path}}/${KEY_FILE}"
 CHAIN_PATH="{{pkg.svc_data_path}}/${CHAIN_FILE}"
-CERT_ID="\\\\VED\\\\Policy\\\\${CERTIFICATE_ZONE}\\\\${COMMON_NAME}"
-
-HOSTNAME="{{sys.hostname}}"
-
-# Set DN paths for device and app creation
-DEVICE_CREATION_DN="\\\\VED\\\\Policy\\\\Devices And Applications\\\\External\\\\${HOSTNAME}"
-APP_CREATION_DN="\\\\VED\\\\Policy\\\\Devices And Applications\\\\External\\\\${HOSTNAME}\\\\${COMMON_NAME}-{{sys.ip}}{{sys.hostname}}"
-
-RUN_ONCE=true
+ID_PATH="{{pkg.svc_data_path}}/${ID_FILE}"
 
 # Establish a CA
 SSL_CERT_FILE="{{pkgPathFor "core/cacerts"}}/ssl/certs/cacert.pem"
 export SSL_CERT_FILE
 
+# https://support.venafi.com/hc/en-us/articles/217991528-Introducing-VCert-API-Abstraction-for-DevOps
 function enroll_new_cert() {
-    # Enroll a new SSL cert
-    exec vcert enroll -no-prompt \
-        -tpp-url ${TPP_URL} \
-        -tpp-user ${TPP_USER} \
-        -tpp-password ${TPP_PASSWORD} \
-        -cert-file ${CERT_PATH} \
-        -key-file ${KEY_PATH} \
-        -chain-file ${CHAIN_PATH} \
-        -cn ${COMMON_NAME} \
-        -z ${CERTIFICATE_ZONE}
+    exec vcert enroll -no-prompt {{#if cfg.cloud.auth.apikey ~}} -k ${CLOUD_APIKEY} {{/if ~}}{{#if cfg.tpp.auth.url ~}} -tpp-user ${TPP_USER} -tpp-password ${TPP_PASSWORD} -u ${TPP_URL} {{/if ~}} -z "{{ cfg.zone }}" -cert-file ${CERT_PATH} -key-file ${KEY_PATH} -chain-file ${CHAIN_PATH} -cn ${COMMON_NAME} -pickup-id-file ${ID_PATH} {{#if cfg.tpp.device.register ~}}-instance ${INSTANCE} -replace-instance -app-info ${APP_INFO} -tls-address ${TLS_ADDRESS}{{/if ~}}
```

Comment:
Would this be better written as:
```
{{#if ...}}
exec vcert enroll ... (cloud specific)
{{/if~}}
{{#if ...}}
exec vcert enroll ... (tpp specific)
{{/if~}}
```
Would that avoid the issue where the echo is needed?

Reply:
... or as:
```
exec vcert enroll \
{{#if ...}} -k ${CLOUD_APIKEY} {{/if ~}} \
```